ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlArrayDecoder.kt

```diff
@@ -99,4 +99,21 @@ public class TomlArrayDecoder(
 
     // this should be applied to [currentPrimitiveElementOfArray] and not to the [rootNode]
     override fun decodeNotNullMark(): Boolean = currentPrimitiveElementOfArray !is TomlNull
+
+    public companion object {
+        /**
+         * @param deserializer - deserializer provided by Kotlin compiler
+         * @param tomlKeyValueArray - TomlKeyValueArray node for decoding
+         * @param config - decoding configuration for parsing and serialization
+         * @return decoded (deserialized) object of type T
+         */
+        public fun <T> decode(
```

Comment:
hm, don't get the idea - why this should be added to particular desrializers like ArrayDecoder?

Reply:
I did the same as I did for map: https://github.com/orchestr7/ktoml/pull/291#discussion_r1988002032
Now in Toml.kt we have:

```kotlin
    private fun <T> decode(deserializer: DeserializationStrategy<T>, parsedToml: TomlFile): T =
        when (deserializer.descriptor.kind) {
            StructureKind.LIST -> TomlArrayDecoder.decode(deserializer, parsedToml.getFirstChild() as TomlKeyValueArray, inputConfig)
            StructureKind.MAP -> TomlMapDecoder.decode(deserializer, parsedToml, inputConfig)
            else -> TomlMainDecoder.decode(deserializer, parsedToml, inputConfig)
        }

```

Reply:
got it!